src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/EETypeNode.cs

```diff
@@ -126,7 +126,7 @@ public virtual void AppendMangledName(NameMangler nameMangler, Utf8StringBuilder
 
         public static bool IsTypeNodeShareable(TypeDesc type)
         {
-            return type.IsParameterizedType || type.IsFunctionPointer || type is InstantiatedType;
+            return type.IsParameterizedType || type.IsFunctionPointer || type is InstantiatedType || type is CompilerGeneratedType;
```

Comment:
We shouldn't be generating an EEType for the compiler generated type. Can you check who is making one?